packages/core/database/lib/entity-manager/relationsOrderer.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const _ = require('lodash/fp');
+
+// TODO: Document this class properly
+/* Constraints:
+  - Expects you will never connect a relation before / after one that does not exist
+  - Expect the last initArr element to be the last relation in the database
+*/
+class FractionalOrderer {
+  constructor(initArr, idColumn, orderColumn) {
+    this.arr = _.castArray(initArr || []).map((r) => ({
+      init: true,
+      id: r[idColumn],
+      order: r[orderColumn],
+    }));
+  }
+
+  _updateRelationOrder(r) {
+    let idx;
+    if (r.position.before) {
+      const { idx: _idx, relation } = this.findRelation(r.position.before);
+      if (relation.init) r.order = relation.order - 0.5;
```

Comment:
Couldn't this potentially break i.e. if your order is 4 but then you update the order twice you're now 3 but something else may think it's order 3?

Would it be better to compare the order value of the next & previous relation, find the diff in order value, half that diff and add / subtract from the prev / next relation (respectively)?

Reply:
Good point!  Let me know if I understand your point correctly:

There is a problem with checking the order value of the next&previous. It is not guaranteed you will have them loaded in memory (as it is paginated). So you do not have that information in here.

Still, I think that use case it is covered here. The connect array is read sequentially so that would result in:
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/20578351/197523103-5c6698cf-2108-4c69-bd4a-8da0b6dd5174.png">


At this point we  we insert the new rows in the database and reorder everything to be integer values.

Reply:
Okay yeah, having seen the 5th column, this makes more sense I didn't realise you then recalculated.

Reply:
yes! it saves a little bit of computing doing it at the end.